src/test/java/com/rishabh/restframework/listeners/TestListener.java

```diff
@@ -36,30 +36,36 @@ public static void setTestMethodName(String name) {
 	public void onTestFailure(ITestResult result) {
 		log.error("The name of the testcase failed is : " + result.getName());
 		extentTest.get().fail(result.getThrowable());
+		extentTest.get().log(Status.FAIL, getTestMethodName() + " Test is failed " + result.getThrowable());
 
 	}
 
 	// When Test case get Skipped, this method is called.
 	@Override
 	public void onTestSkipped(ITestResult result) {
 		log.debug("The name of the testcase Skipped is : " + result.getName());
+		extentTest.get().log(Status.SKIP, result.getName() + " Test is Skipped" +  result.getThrowable());
 	}
 
 	// When Test case get Started, this method is called.
 	@Override
 	public void onTestStart(ITestResult result) {
 		log.info("The name of test case started is : " + result.getName());
-		setTestMethodName(result.getMethod().getMethodName());
+		if (result.getMethod().getDescription() == null)
+			setTestMethodName(result.getMethod().getMethodName());
+		else
+			setTestMethodName(result.getMethod().getDescription());
```

Comment:
Reference it Will set the TestCaseDescription in Test Reports :
<img width="692" alt="Screenshot 2021-05-27 at 3 12 11 PM" src="https://user-images.githubusercontent.com/38694449/119804657-2ebc6000-befe-11eb-8f05-af1dc0e13c7c.png">